doc/source/migration.rst

```diff
@@ -0,0 +1,105 @@
+
+To call ``asyncio`` code from Eventlet code, you can wrap an coroutine (or anything you can ``await``) into an Eventlet ``GreenThread``.
+For example, if you want to a HTTP request from Eventlet, you can use the ``asyncio``-based ``aiohttp`` library::
+
+    import aiohttp
+    from eventlet.asyncio import spawn_for_awaitable
+
+    async def request():
+        async with aiohttp.ClientSession() as session:
+            url = "https://example.com"
+            async with session.get(url) as response:
+                html = await response.text()
+                return html
+
+
+    # This makes a coroutine; typically you'd ``await`` it:
+    coro = request()
+
+    # You can wrap this coroutine with an Eventlet GreenThread, similar to
+    # ``evenlet.spawn()``:
+    gthread = spawn_for_awaitable(request())
+
+    # And then get its result, the body of https://example.com:
+    result = gthread.wait()
+
+In the other direction, any ``eventlet.greenthread.GreenThread`` can be ``await``-ed in ``async`` functions.
+In other words ``async`` functions can call into Eventlet code::
+
+    def blocking_eventlet_api():
+        eventlet.sleep(1)
+        # do some other pseudo-blocking work
+        # ...
+        return 12
+
+    async def my_async_func():
+        gthread = eventlet.spawn(blocking_eventlet_api)
+        # In normal Eventlet code we'd call gthread.wait(), but since this is an
+        # async function we'll want to await instead:
+        result = await gthread
+        # result is now 12
+        # ...
+
+Cancellation of ``asyncio.Future`` and killing of ``eventlet.GreenThread`` should propagate between the two.
+
+Using these two APIs, with more to come, you can gradually migrate portions of your application or library to ``asyncio``.
+Calls to blocking APIs like ``urlopen()`` or ``requests.get()`` can get replaced with calls to ``aiohttp``, for example.
+
+
+Step 3. Drop Eventlet altogether
+--------------------------------
+
+Eventually you won't be relying on Eventlet at all: all your code will be ``asyncio``-based.
+At this point you can drop Eventlet and switch to running the ``asyncio`` loop directly.
+
+
+Known limitations and work in progress
+--------------------------------------
+
+In general, ``async`` functions and Eventlet green threads are two separate universes that just happen to be able to call each other.
+In ``async`` functions:
+
+* Eventlet thread locals probably won't work correctly.
+* ``evenlet.greenthread.getcurrent()`` won't give the result you expect.
+* ``eventlet`` locks and queues won't work if used directly.
+
+In Eventlet greenlets:
+
+* ``asyncio`` locks won't work if used directly.
+
+We expect to add more migration and integration APIs over time as we learn more about what works, common idioms, and requirements for migration.
+You can try progress in the `GitHub issue <https://github.com/eventlet/eventlet/issues/868>`_, and file new issues if you have problems.
```

Comment:
This sentence looks weird to me, especially "You can try progress in the". Did you mean track your progress?